datagrepper/app.py

```diff
@@ -131,74 +134,35 @@ def remove_session(exc):
     dm.session.remove()
 
 
-def modify_rst(rst):
-    """Downgrade some of our rst directives if docutils is too old."""
-
-    try:
-        # The rst features we need were introduced in this version
-        minimum = [0, 9]
-        version = map(int, docutils.__version__.split("."))
-
-        # If we're at or later than that version, no need to downgrade
-        if version >= minimum:
-            return rst
-    except Exception:
-        # If there was some error parsing or comparing versions, run the
-        # substitutions just to be safe.
-        pass
-
-    # Otherwise, make code-blocks into just literal blocks.
-    substitutions = {
-        ".. code-block:: javascript": "::",
-    }
-    for old, new in substitutions.items():
-        rst = rst.replace(old, new)
-
-    return rst
-
-
-def modify_html(html):
-    """Perform style substitutions where docutils doesn't do what we want."""
-
-    substitutions = {
-        '<tt class="docutils literal">': "<code>",
-        "</tt>": "</code>",
-    }
-    for old, new in substitutions.items():
-        html = html.replace(old, new)
-
-    return html
-
-
 def preload_docs(endpoint):
     """Utility to load an RST file and turn it into fancy HTML."""
 
     here = os.path.dirname(os.path.abspath(__file__))
     default = os.path.join(here, "docs")
     directory = app.config.get("DATAGREPPER_DOC_PATH", default)
-    fname = os.path.join(directory, endpoint + ".rst")
+    args = " -b html -c %s %s %s/_build/html/" % (default, default, default)
+    sphinx_main(args.split())
+    htmldocs = os.path.join(directory, "_build/html")
+    fname = os.path.join(htmldocs, endpoint + ".html")
     with codecs.open(fname, "r", "utf-8") as f:
-        rst = f.read()
-
-    rst = modify_rst(rst)
-
-    api_docs = docutils.examples.html_body(rst)
-
-    api_docs = modify_html(api_docs)
+        api_docs = f.read()
 
+    soup = bs4(api_docs, "html.parser")
+    api_docs = soup.find("body").decode_contents()
+    
     api_docs = markupsafe.Markup(api_docs)
     return api_docs
 
 
-htmldocs = dict.fromkeys(["index", "reference", "widget", "charts"])
+htmldocs = dict.fromkeys(["index", "reference", "widget", "charts", "contributing"])
+
 for key in htmldocs:
     htmldocs[key] = preload_docs(key)
```

Comment:
This looks like it's calling `preload_docs()`, and thus `sphinx_main` for each endpoint. It is unnecessary as `sphinx_main` will build all the endpoints in a single run.